### DISCUSSION TITLE
How to  train the model with my own letter images

### DISCUSSION THREAD

**Discussion by Shiva-Ai4m** [mindee/doctr#1247] [General]
How i can train the model with own letter images for better recognitions ? 
I have a letter images which I need to train and put it to recognition.
@felixT2K Can you  please help me out in this?

**Comment by felixT2K** [comment]
Hi @Shivashankarar :wave:,

If i understood it correctly each image contains only one letter ?

**Reply by Shiva-Ai4m** [reply]
yes, for training the model vocabulary. so I think  it will be better for recognition

**Reply by felixT2K** [reply]
Ok short recap:

What vocabulary are we talking about and what model would you like to train?
In the end, the model should be able to recognize words and not just individual letters correctly?

**Reply by Shiva-Ai4m** [reply]
i would like to train crnn_vgg16_bn model and i need to recognize the words correctly 
so my ideas was if I have the letters then the model can get it and read it correctly

**Reply by felixT2K** [reply]
Ok now I understand what you are trying to do. 
If you want to recognize words you need to provide a dataset which contains word crop images and corresponding labels.
(ref.: https://github.com/mindee/doctr/tree/main/references/recognition)
For single letter recognition it would be enough to train a classification model. (one of our backbones)
About which vocab do we talk ?

**Reply by Shiva-Ai4m** [reply]
yes, the vocab which we talk

**Comment by felixT2K** [comment]
> yes, the vocab which we talk

Which ? ^^  english, german, french, .. ?

**Reply by Shiva-Ai4m** [reply]
english

**Reply by felixT2K** [reply]
Ok but for english you could use the pretrained french model because the french vocab contains all english letters/numbers/symbols. 
Does it not work well for you ?

**Reply by Shiva-Ai4m** [reply]
i mean it is working for me but I think if I train the model with the font on images which I need to recognize the textit will be able to detect the text better 
Example 
I have a image in which the text is 
"10g H24A JAN.23 JUN.23"
But it is detected as 
109 H24A JAN.23 JUN.23
I am attaching the a part of whole image in this. 
![235641149-aa9120aa-946b-46f0-8073-07ca29f04054](https://github.com/mindee/doctr/assets/122434059/bd6cdd3c-9d13-479b-8513-44114e4a8b02)

**Reply by felixT2K** [reply]
Ah i see the `g` is not recognized correctly (but is really hard to read 😅 ). 
I don't think that `crnn` will recognize such cases better.
Which backend do you use ? PyTorch or TensorFlow ?

**Reply by Shiva-Ai4m** [reply]
tensorflow

**Reply by felixT2K** [reply]
We will add today or tomorrow a pretrained version for `master` model in tensorflow. This should be more possible to recognize also some side cases correctly maybe.
